lib/sqlalchemy/ext/orderinglist.py

```diff
@@ -330,13 +336,13 @@ def _order_entity(self, index, entity, reorder=True):
         if have is not None and not reorder:
             return
 
-        should_be = self.ordering_func(index, self)
+        should_be = self.ordering_func(index_to_int(index), self)
         if have != should_be:
             self._set_order_value(entity, should_be)
 
-    def append(self, entity):
-        super().append(entity)
-        self._order_entity(len(self) - 1, entity, self.reorder_on_append)
+    def append(self, __entity: _T) -> None:
```

Comment:
~~you can use `/` to indicate a positional only argument.~~

never-mind, that's 2.1 only